lib/facility.rb

```diff
@@ -1,14 +1,85 @@
+require 'pry'
+require 'pry-reload'
+# require './lib/vehicle'
+# require './lib/registrant'
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  
+  # Only Enables Writing (Setting New Values) of Properties, Not Reading
+  # attr_writer :name, :address, :phone, :services
+
+  # Enables Both Reading / Writing Properties
+  # attr_accessor :name, :address, :phone, :services
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  # Only Enables Reading of Properties, Not Writing (Setting New Values)
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services, 
+              :registered_vehicles, 
+              :collected_fees
+
+  def initialize(facility_obj)
+    @name = facility_obj[:name]
+    @address = facility_obj[:address]
+    @phone = facility_obj[:phone]
     @services = []
+    @registered_vehicles = []
   end
 
-  def add_services(service)
+  def add_service(service)
+    @collected_fees = 0
```

Comment:
I'd suggest defaulting the @collected_fees in the initialize rather than here. Every time you add a service, this instance var would get re-assigned back to 0 which is not expected behavior. If you default it in the initialize, however, it can accurately store all the collected fees for any service!